posts/2020-12-18-microprofile-long-running-actions-beta.adoc

```diff
@@ -0,0 +1,400 @@
+---
+layout: post
+title: "MicroProfile Long Running Actions in Open Liberty"
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/yongja79
+author_github: https://github.com/yongja79
+seo-title: MicroProfile Long Running Actions in Open Liberty
+seo-description: Try MicroProfile Long Running Actions in the Open Liberty 20.0.0.12-beta
+blog_description: "Try MicroProfile Long Running Actions in the Open Liberty 20.0.0.12-beta"
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= MicroProfile Long Running Actions in Open Liberty
+Jason Yong <https://github.com/yongja79>
+
+In the 20.0.0.12-beta release of Open Liberty a new capability was introduced to support loosely-coupled transaction semantics for Java microservice. This is an implementation of a new MicroProfile specification for Long Running Actions (MicroProfile LRAs).  
+
+== What is a MicroProfile LRA?
+
+MicroProfile LRA is a MicroProfile specification that provides a simple, loosely-coupled transaction model for microservices and is based on the Saga pattern for distributed transactions . It relaxes some of the constraints of ACID transactions to enable independent microservices to more easily participate in a potentially long-running orchestrated activity, with each microservice contributing the encapsulated business logic required to align with the overall outcome of the activity or "transaction". 
+
+== So how does MicroProfile LRA help? 
```

Comment:
== Why do we need MicroProfile LRA?
In traditional applications, the well-known transaction model is 2-phase commit. This introduces resource locking and global rollback, which does not scale well and therefore not suitable for microservice architecture. As a consequence, [SAGA pattern](https://developer.ibm.com/depmodels/microservices/articles/use-saga-to-solve-distributed-transaction-management-problems-in-a-microservices-architecture#saga) was defined for microservice transaction architecture. A SAGA is a sequence of transactions. If one transaction fails, the saga undoes all of the preceding transactions. It aims for eventual consistence. MicroProfile LRA is an implementation of the SAGA pattern.